docs/settings api.md

```diff
@@ -0,0 +1,187 @@
+# Archipelago Settings API
+
+The settings API describes how to use installation-wide config and let the user configure them, like paths, etc. using
+host.yaml. For the player settings / player yamls see [options api.md](options api.md).
+
+The settings API replaces `Utils.get_options()` and `Utils.get_default_options()`
+as well as the predefined `host.yaml` in the repository.
+
+For backwards compatibility with APWorlds, some interfaces are kept for now and will produce a warning when being used.
+
+
+## Config File
+
+Settings use options.yaml (manual override), if that exists, or host.yaml (the default) otherwise.
+The files are searched for in the current working directory, if different from install directory, and in `user_path`,
+which either points to the installation directory, if writable, or to %home%/Archipelago otherwise.
+
+**Examples:**
+* C:\Program Data\Archipelago\options.yaml
```

Comment:
I don't touch filenames in this PR because migrating from one filename to another is gonna be easier once that's in and breaking someone's options.yaml will make sad faces.